apps/server/src/resources/cluster/business.ts

```diff
@@ -25,6 +44,184 @@ export const getAllUserClusters = async (kcUser: UserProfile) => {
   const clusters = await listClustersForUser(where)
   return clusters.map(({ stages, ...cluster }) => ({
     ...cluster,
-    stageIds: stages.map(({ id }) => id) ?? [],
+    stageIds: stages?.map(({ id }) => id) ?? [],
   }))
 }
+
+export const checkClusterProjectIds = (data: Omit<ClusterDetails, 'id'> & { id?: Cluster['id'] }) => {
```

Comment:
```suggestion
export const getClusterProjectIds = (data: Omit<ClusterDetails, 'id'> & { id?: Cluster['id'] }) => {
```
Le check est trompeur car ça ne vérifie pas grand chose je trouve